osu.Game/Screens/Play/MasterGameplayClockContainer.cs

```diff
@@ -197,6 +207,51 @@ public void StopUsingBeatmapClock()
             addAdjustmentsToTrack();
         }
 
+        protected override void Update()
+        {
+            base.Update();
+            checkPlaybackValidity();
+        }
+
+        #region Clock validation (ensure things are running correctly for local gameplay)
+
+        private double elapsedGameplayClockTime;
+        private double? elapsedValidationTime;
+        private int playbackDiscrepancyCount;
+
+        private const int allowed_playback_discrepancies = 5;
+
+        private void checkPlaybackValidity()
+        {
+            if (GameplayClock.IsRunning)
+            {
+                elapsedGameplayClockTime += GameplayClock.ElapsedFrameTime;
+
+                elapsedValidationTime ??= elapsedGameplayClockTime;
+                elapsedValidationTime += GameplayClock.Rate * Time.Elapsed;
+
+                if (Math.Abs(elapsedGameplayClockTime - elapsedValidationTime!.Value) > 300)
+                {
+                    if (playbackDiscrepancyCount++ > allowed_playback_discrepancies)
+                    {
+                        if (playbackRateValid.Value)
+                        {
+                            playbackRateValid.Value = false;
+                            Logger.Log("System audio playback is not working as expected. Some online functionality will not work.\n\nPlease check your audio drivers.", level: LogLevel.Important);
```

Comment:
Should probably be a localizable notification, as  this is a user-facing string.

Reply:
it's an edge case that only 1 in 10000000 will see, but sure, at some opint